tests/v3_api/test_secrets.py

```diff
@@ -0,0 +1,331 @@
+from common import *
+import pytest
+import base64
+
+namespace = {"p_client": None, "ns": None, "cluster": None,  "project": None}
+
+
+def test_secret_create_all_ns():
+
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    name = random_test_name("default")
+    keyvaluepair = {"testall": "valueall"}
+
+    secret = p_client.create_secret(name=name, namespaceId='NULL',
+                                    stringData=keyvaluepair)
+    print secret
+
+    assert secret['baseType'] == "secret"
+    secretdata = secret['data']
+
+    for k, v in secretdata.items():
+        key = k
+        vdecoded = base64.b64decode(v)
+
+    assert key == keyvaluepair.keys()[0]
+    assert vdecoded == keyvaluepair.values()[0]
+
+    # Create Workload with secret as volume
+    mountpath = "/test"
+    volumeMounts = [{"readOnly": False, "type": "volumeMount",
+                     "mountPath": mountpath, "name": "vol1"}]
+    con = [{"name": "test1",
+            "image": "sangeetha/testclient",
+            "volumeMounts": volumeMounts}]
+
+    secretId = secret['name']
+
+    volumes = [{"type": "volume", "name": "vol1",
+                "secret": {"type": "secretVolumeSource", "defaultMode": 256,
+                           "secretId": secretId,
+                           "optional": False, "items": "NULL"}}]
+
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id, volumes=volumes)
+    print workload
+    # Validate Workload with secret
+    validate_workload_with_secret(p_client, workload, "deployment",
+                                  ns.name, keyvaluepair)
+
```

Comment:
Can you include in this  test the use case of creating a new namespace in this project and deploying workload with secrets that already exists in this namespace?

Reply:
Created new namespace and deployed workload